pandas/tests/reshape/test_pivot.py

```diff
@@ -2153,3 +2153,37 @@ def test_pivot_index_none(self):
 
         expected.columns.name = "columns"
         tm.assert_frame_equal(result, expected)
+
+    def test_pivot_index_list_values_none_modifies_args(self):
+        # GH37635
+        df = DataFrame(
+            {
+                "lev1": [1, 1, 1, 2, 2, 2],
+                "lev2": [1, 1, 2, 1, 1, 2],
+                "lev3": [1, 2, 1, 2, 1, 2],
+                "lev4": [1, 2, 3, 4, 5, 6],
+                "values": [0, 1, 2, 3, 4, 5],
+            }
+        )
+        index = ["lev1", "lev2"]
+        columns = ["lev3"]
+        pivoted = df.pivot(index=index, columns=columns, values=None)
+
+        expected = DataFrame(
+            {
+                "a": [1.0, 3.0, 5.0, float("nan")],
+                "b": [2.0, float("nan"), 4.0, 6.0],
+                "c": [0.0, 2.0, 4.0, float("nan")],
+                "d": [1.0, float("nan"), 3.0, 5.0],
+            }
+        )
+        expected.index = MultiIndex.from_arrays(
```

Comment:
put the index and column names into the data frame constructor instead of setting them like this (can pass them in as keyword args)